File: src/main/java/de/gmasil/example/ExampleFrame.java
package de.gmasil.example;

import javax.swing.JFrame;
import java.awt.event.WindowAdapter;
import java.awt.event.WindowEvent;

public class ExampleFrame extends JFrame {
    public ExampleFrame() {
        super("Example App");
        setSize(250, 100);
        setVisible(true);
        addWindowListener(new WindowAdapter() {
            @Override
            public void windowClosing(WindowEvent e) {
                dispose();
            }
        });
    }
}

File: src/main/kotlin/de/gmasil/example/ExampleFrame.kt
package de.gmasil.example

import java.awt.event.WindowAdapter
import java.awt.event.WindowEvent
import javax.swing.JFrame

class ExampleFrame() : JFrame("Example App") {
    init {
        setSize(250, 100)
        isVisible = true
        addWindowListener(object : WindowAdapter() {
            override fun windowClosing(we: WindowEvent) {
                dispose()
            }
        })
    }
}
